File: content/tracks/dsa/problems/lc-0121-best-time-to-buy-and-sell-stock/Solution.java
import java.util.*;

class Solution {
    public int maxProfit(int[] prices) {
        // One pass: remember the cheapest day so far and try selling on each day.
        int cheapest = prices[0];
        int best = 0;
        for (int price : prices) {
            cheapest = Math.min(cheapest, price);
            best = Math.max(best, price - cheapest);
        }
        return best;
    }
}

File: content/tracks/dsa/problems/lc-0121-best-time-to-buy-and-sell-stock/solution.py
from typing import List


class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        # One pass: remember the cheapest day so far and try selling on each day.
        cheapest = prices[0]
        best = 0
        for price in prices:
            cheapest = min(cheapest, price)
            best = max(best, price - cheapest)
        return best
